src/libAtomVM/popcorn/popcorn_nifs.c

```diff
@@ -559,103 +557,123 @@ static term nif_ets_new(Context *ctx, int argc, term argv[])
     term private = interop_kv_get_value(options, ATOM_STR("\x7", "private"), ctx->global);
     term public = interop_kv_get_value(options, ATOM_STR("\x6", "public"), ctx->global);
 
-    PopcornEtsAccessType access = PopcornEtsAccessProtected;
+    Popcorn2EtsTableAccess access = Popcorn2EtsTableAccessProtected;
     if (!term_is_invalid_term(private)) {
-        access = PopcornEtsAccessPrivate;
+        access = Popcorn2EtsTableAccessPrivate;
     } else if (!term_is_invalid_term(public)) {
-        access = PopcornEtsAccessPublic;
+        access = Popcorn2EtsTableAccessPublic;
     }
 
-    PopcornEtsTableType type = PopcornEtsTableSet;
-    term is_duplicate_bag = interop_kv_get_value_default(options, ATOM_STR("\xd", "duplicate_bag"), FALSE_ATOM, ctx->global) == TRUE_ATOM;
-    if (is_duplicate_bag) {
-        type = PopcornEtsTableDuplicateBag;
+    term bag = interop_kv_get_value(options, ATOM_STR("\x3", "bag"), ctx->global);
+    term duplicate_bag = interop_kv_get_value(options, ATOM_STR("\xd", "duplicate_bag"), ctx->global);
+
+    Popcorn2EtsTableType type = Popcorn2EtsTableSet;
+    if (!term_is_invalid_term(bag)) {
+        type = Popcorn2EtsTableBag;
+    } else if (!term_is_invalid_term(duplicate_bag)) {
+        type = Popcorn2EtsTableDuplicateBag;
```

Comment:
🌵 I feel like would be nice to note in comment that `ets` can have both in options and we explicitly have priority on `duplicate_bag`.